### ISSUE TITLE
Initial master push

### ISSUE THREAD

**Issue by xintron** [xintron/zpam#1]
Defines what should be done before the core branch is merged into master.

- [ ] Backend interface.
- [ ] Identifier struct - common structure to be used for backend implementations
- [x] Config system - handle bot config and allow plugins to share the config file with easy dispatch to concerned plugins
- [ ] Plugin interface - add register handler and solution for plugin instantiation.

Backends to implement
- [ ] Shell - interactive terminal to use when developing plugins
- [ ] IRC
- [ ] Slack
- [ ] Hipchat

Documentation
- [ ] Fully documented public functions
- [ ] README - contain example of how to easily run the bot

**Comment by cromigon** [comment]
Are you considering any "core plugins" to be shipped together with the first master merge, or is it something that should come at a later date or only be featured in the documentation?

I thought to ask seeing as you mentioned plugins under storage, config and documentation.

**Comment by xintron** [comment]
@cromigon There will probably be a few example plugins shipped with the bot initially to showcase how plugins can be built. Most of the plugin development will be described in the Wiki and briefly in the readme.
